core/store/src/lib.rs

```diff
@@ -406,6 +279,19 @@ impl StoreUpdate {
     }
 }
 
+impl std::ops::Deref for StoreUpdate {
+    type Target = DBTransaction;
+    fn deref(&self) -> &Self::Target {
+        &self.transaction
+    }
+}
+
+impl std::ops::DerefMut for StoreUpdate {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.transaction
+    }
+}
```

Comment:
Not a huge fan of using deref over just manually delegating. This priorities writer over reader: the write can just deref and get the API "for free", but the reader now has to piece the StoreUpdate API from StoreUpdate proper and the stuff it deref's to. Deref is a subtle and easy to miss thing. 

Additionally, delegation here would not be pure boilerplate, and can genuinely increase the abstraction level: In DbTx, we can implement core API, with `Vec<u8>` and `increse_refcount_by`, and in the StoreUpdate we can have convenience layer, where we use `Into<Vec<u8>>` and define `increase-refcount` in terms of `increase-refcount-by`. 

Do we actualy need both `DbTx` and `StoreUpdate`? Can we merge them into a single type? 

No super-strong opinion here.